From: Hannes Reinecke <hare@suse.de>
Date: Tue, 17 Jul 2018 16:04:32 +0200
Subject: [PATCH] nvme: kABI fixes for nvmet_ctrl
References: bsc#1054245
Patch-Mainline: never, kABI fix

The latest update to add ns_changed_log and AEN configuration broke
the kABI, so insert GENKSYMS guards and reshuffle the added fields.

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 drivers/nvme/target/nvmet.h | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/nvme/target/nvmet.h b/drivers/nvme/target/nvmet.h
index 8b02451bb38d..6688b574c8af 100644
--- a/drivers/nvme/target/nvmet.h
+++ b/drivers/nvme/target/nvmet.h
@@ -135,10 +135,6 @@ struct nvmet_ctrl {
 	u16			cntlid;
 	u32			kato;
 
-	struct nvmet_port	*port;
-
-	u32			aen_enabled;
-	unsigned long		aen_masked;
 	struct nvmet_req	*async_event_cmds[NVMET_ASYNC_EVENTS];
 	unsigned int		nr_async_event_cmds;
 	struct list_head	async_events;
@@ -149,11 +147,15 @@ struct nvmet_ctrl {
 
 	struct nvmet_fabrics_ops *ops;
 
-	__le32			*changed_ns_list;
-	u32			nr_changed_ns;
-
 	char			subsysnqn[NVMF_NQN_FIELD_LEN];
 	char			hostnqn[NVMF_NQN_FIELD_LEN];
+#ifndef __GENKSYMS__
+	struct nvmet_port	*port;
+	u32			aen_enabled;
+	unsigned long		aen_masked;
+	__le32			*changed_ns_list;
+	u32			nr_changed_ns;
+#endif
 };
 
 struct nvmet_subsys {
-- 
2.12.3

